Tetris/main.js

```diff
@@ -0,0 +1,438 @@
+'use strict';
+
+const { MyEventEmitter, hideLoops } = require('./extraConstructions.js');
+
+const eventEm = new MyEventEmitter();
+
+const typeFigures = [
+  [
+    [0, 1, 0],
+    [1, 1, 1],
+    [0, 0, 0]
+  ],
+  [
+    [1, 1, 0],
+    [1, 1, 0],
+    [0, 0, 0]
+  ],
+  [
+    [1, 1, 0],
+    [0, 1, 1],
+    [0, 0, 0]
+  ],
+  [
+    [1, 1, 1],
+    [1, 0, 0],
+    [0, 0, 0]
+  ],
+  [
+    [1, 1, 1, 1],
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+    [0, 0, 0, 0]
+  ]
+];
+
+
+const shapeFigures = [
+  '\x1b[31m[]\x1b[0m',  // RED
+  '\x1b[32m[]\x1b[0m',  // Green
+  '\x1b[33m[]\x1b[0m',  // Yellow
+  '\x1b[34m[]\x1b[0m',  // Blue
+  '\x1b[35m[]\x1b[0m',  // Purple
+];
+
+
+// Повертає ціле значення від 0 до 4
+// (використовується для shapeFigures i typeFigures)
+const randomFrom0To4 = () => {
+  const result = Math.floor((Math.random() * 5));
+  return result;
+};
```

Comment:
А еще лучше сделать функции `random(from, to)` и потом из нее подставлением константы 4, и 4 нужно в константу.
```suggestion
const randomFrom0To4 = () => Math.floor(Math.random() * 5);
```
Именование тоже не лучшее.